sdk/storage/Azure.Storage.Blobs.DataMovement/src/BlobScanner.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Specialized;
+
+namespace Azure.Storage.Blobs.DataMovement
+{
+    /// <summary>
+    /// AzureContainerScanner class.
+    /// </summary>
+    public class BlobScanner
+    {
+        private readonly BlobContainerClient _containerClient;
+
+        /// <summary>
+        /// Constuctor for AzureContainerScanner.
+        /// </summary>
+        /// <param name="client"></param>
+        public BlobScanner(BlobContainerClient client)
+        {
+            // Hacky deep-clone to prevent client changes mid-operation
+            _containerClient = client.GetParentBlobServiceClient().GetBlobContainerClient(client.Name);
```

Comment:
Is there any better way to do this?

Is it necessary? Are any public fields/fields modifiable through methods actually able to interfere with client functionality, or does functionality only reference the private fields set by ctor?

Reply:
I might have to make an internal `DeepCopy` to call so we don't have to do something like this.

I will submit a PR hopefully today so you can pull and replace this.

Reply:
Alright, I'll just store it as given then.